safehttp/doc.go

```diff
@@ -12,5 +12,113 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Package safehttp TODO(@kele, @empijei)
+// Package safehttp provides a framework for building secure-by-default HTTP
+// servers. See https://github.com/google/go-safeweb#readme to learn about the
+// goals and features.
+//
+// Safe Responses
+//
+// HTTP responses need to be crafted carefully in order to prevent common web
+// vulnerabilities like Cross-site Scripting (XSS). To help with this, we use
+// Safe Responses.
+//
+// Safe Responses are HTTP responses which has been determined to be safe when
+// received by a modern, popular web browser. For example, we consider HTML
+// responses generated using a template system enforcing contextual autoescaping
+// to be safe e.g. modern Angular, github.com/google/safehtml/template).
+//
+// The Go Safe Web ResponseWriter implementation accepts exclusively Safe
+// Responses, instead of byte slices.
+//
+// Since different projects will consider different ways of crafting a response
+// safe, we offer a way of configuring this in the framework. Whether a response
+// is considered safe or not is determined by the Dispatcher.
+//
+// Dispatcher
+//
+// An implementation of a Dispatcher should be provided by security experts in
+// your project. The Dispatcher is called for every write method of the
+// ResponseWriter and is used to determine whether the passed response should be
+// considered safe. The Dispatcher is responsible for writing the response to
+// the underlying http.ResponseWriter in a safe way.
```

Comment:
I'd highlight that the Dispatcher might contain vulnerabilities and that the security of the overall project is relying on a sound Dispatcher implementation

Reply:
Added a Warning a few lines below.